internal/entities/liquidity_provider/liquidity_provider.go

```diff
@@ -117,3 +121,28 @@ type DefaultCredentialsSetEvent struct {
 	entities.Event
 	Credentials *HashedCredentials
 }
+
+func ValidateConfiguration[T ConfigurationType](
+	displayName string,
+	signer entities.Signer,
+	readFunction func() (*entities.Signed[T], error),
+) (*entities.Signed[T], error) {
+	configuration, err := readFunction()
+	if err != nil {
+		log.Errorf("Error getting %s configuration, using default configuration. Error: %v", displayName, err)
+		return nil, err
+	}
+	if configuration == nil {
+		log.Warnf("Custom %s configuration not found. Using default configuration.", displayName)
+		return nil, errors.New("configuration not found")
+	}
+	if err = configuration.CheckIntegrity(crypto.Keccak256); err != nil {
```

Comment:
you should avoid this, you're importing a function from ethereum package in the entity layer which shouldn't use libraries at all (we can make an exception with the log), to avoid this you should receive the hash function as a parameter

Reply:
Changed.